File: oss/src/main/java/net/yan100/compose/oss/abstracts/InMap.java
package net.yan100.compose.oss.abstracts;

import java.io.InputStream;


/**
 * 文件输入参数构造器
 *
 * @author TrueNine
 * @since 2023-02-20
 */
public interface InMap extends StreamsMap {
  /**
   * @return 使用的输入流
   */
  InputStream usedStream();

  /**
   * @return 媒体类型
   */
  String mimeType();

  /**
   * @return 文件名
   */
  String fileName();

  /**
   * @return 目录名
   */
  String directoryName();

  long size();
}

File: oss/src/main/kotlin/net/yan100/compose/oss/InMap.kt
package net.yan100.compose.oss

import java.io.InputStream

/**
 * 文件输入参数构造器
 *
 * @author TrueNine
 * @since 2023-02-20
 */
interface InMap : StreamsMap {
  override val usedStream: InputStream
  override val mediaType: String
  override val fName: String
  override val dirName: String
  override val size: Long
}
